core/trino-main/src/main/java/io/trino/connector/CoordinatorDynamicCatalogManager.java

```diff
@@ -286,6 +289,93 @@ public void createCatalog(CatalogName catalogName, ConnectorName connectorName,
         }
     }
 
+    @Override
+    public void renameCatalog(CatalogName catalogName, CatalogName newCatalogName)
+    {
+        requireNonNull(catalogName, "catalogName is null");
+
+        catalogsUpdateLock.lock();
+        try {
+            checkState(state != State.STOPPED, "ConnectorManager is stopped");
+
+            if (!activeCatalogs.containsKey(catalogName)) {
+                throw new TrinoException(NOT_FOUND, format("Catalog '%s' does not exist", catalogName));
+            }
+            if (activeCatalogs.containsKey(newCatalogName)) {
+                throw new TrinoException(ALREADY_EXISTS, format("Catalog '%s' already exists", newCatalogName));
+            }
+
+            createCatalogLikeInternal(catalogName, newCatalogName, ImmutableMap.of());
```

Comment:
Why is rename modeled as "clone the catalog into a new one, drop the old one" vs adding proper support for renaming catalogs? I'm concerned about the impact of any state that needs to be dropped and re-created, caches, etc., that would otherwise not be necessary to manage with a proper rename.

Reply:
Got it. It's not ideal, though. It shouldn't be necessary to re-create the catalog just to give it a new name. 

I guess the underlying problem is that `ConnectorFactory::create` takes a name. We should look into (for a future improvement) what it would take to decouple the instance from its name so that the name is just an entry in the catalog maps.